Add a new --executable option to distribute so that we can
force the shebang line in installed python scripts.

--- a/setuptools/command/install.py
+++ b/setuptools/command/install.py
@@ -16,6 +16,7 @@ class install(orig.install):
     """Use easy_install to install the package, w/dependencies"""
 
     user_options = orig.install.user_options + [
+        ('executable=', 'e', "specify final destination interpreter path"),
         ('old-and-unmanageable', None, "Try not to use this!"),
         ('single-version-externally-managed', None,
          "used by system package builders to create 'flat' eggs"),
@@ -31,6 +32,7 @@ class install(orig.install):
 
     def initialize_options(self):
         orig.install.initialize_options(self)
+        self.executable = None
         self.old_and_unmanageable = None
         self.single_version_externally_managed = None
 
--- a/setuptools/command/install_scripts.py
+++ b/setuptools/command/install_scripts.py
@@ -12,6 +12,13 @@
     def initialize_options(self):
         orig.install_scripts.initialize_options(self)
         self.no_ep = False
+        self.executable = None
+
+    def finalize_options(self):
+        orig.install_scripts.finalize_options(self)
+        self.set_undefined_options('install',
+                ('executable','executable')
+        )
 
     def run(self):
         import setuptools.command.easy_install as ei
@@ -32,6 +39,8 @@
         )
         bs_cmd = self.get_finalized_command('build_scripts')
         exec_param = getattr(bs_cmd, 'executable', None)
+        if self.executable is not None:
+            exec_param = self.executable
         bw_cmd = self.get_finalized_command("bdist_wininst")
         is_wininst = getattr(bw_cmd, '_is_running', False)
         writer = ei.ScriptWriter

